### PR TITLE
Updating XDS protocol details

### PR DESCRIPTION
**Pull request by ramramhariram**
1 - Added a reference to note that it is XDS Protocol 2 - Updated to reflect changes in 1.11 (no longer support v2)  3 - Added a cross reference.

### Description
Describe why you're making this change, in plain English.

### Testing & Reproduction steps
* In the case of bugs, describe how to replicate
* If any manual tests were done, document the steps and the conditions to replicate
* Call out any important/ relevant unit tests, e2e tests or integration tests you have added or are adding

### Links
Include any links here that might be helpful for people reviewing your PR (Tickets, GH issues, API docs, external benchmarks, tools docs, etc). If there are none, feel free to delete this section.

Please be mindful not to leak any customer or confidential information. HashiCorp employees may want to use our internal URL shortener to obfuscate links.

### PR Checklist

* [ ] updated test coverage
* [ ] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

File: website/content/docs/upgrading/compatibility.mdx [hunk 1]
```diff
@@ -30,14 +30,17 @@ at a time, one version at a time.
 
 For more details on the specifics of upgrading, see the [upgrading page](/docs/upgrading).
 
-## Protocol Compatibility Table
+## XDS Protocol Compatibility Table
```
Comment (jkirschner-hashicorp) [inline website/content/docs/upgrading/compatibility.mdx:33]: I _think_ this table is showing Consul agent protocol compatibility, and has nothing to do with Envoy xDS. I'm flagging this for now while I seek clarification from engineering. Regardless, I agree that what "protocol" we're referring to here needs clarification!

**Review by jkirschner-hashicorp — CHANGES_REQUESTED** [review]
Marking "request changes" until accuracy can be checked with engineering. This table could definitely benefit from better clarity. Thanks!

File: website/content/docs/upgrading/compatibility.mdx (hunk 1 above)
Comment (blake) [reply website/content/docs/upgrading/compatibility.mdx:33]: This refers to the RPC protocol version.

https://github.com/hashicorp/consul/blob/b29e5894f2322c9294b1f03c9b264ae2e6ae0eb0/agent/config/config.go#L212

I believe support for xDS v2 was dropped in hashicorp/consul#11354 when that PR dropped support for Envoy 1.16.5 in [Consul 1.11.0](https://github.com/hashicorp/consul/blob/main/CHANGELOG.md#1110-december-14-2021). At the time, all of the more recent, supported versions of Envoy also the newer xDS v3.

File: website/content/docs/upgrading/compatibility.mdx [hunk 2]
```diff
@@ -30,14 +30,17 @@ at a time, one version at a time.
...
 | -------------- | ---------------------------------------------------------------------------- |
 | 0.1 - 0.3      | 1                                                                            |
 | 0.4            | 1, 2                                                                         |
 | 0.5            | 1, 2. 0.5.X servers cannot be mixed with older servers.                      |
 | 0.6            | 1, 2, 3                                                                      |
-| >= 0.7         | 2, 3. Will automatically use protocol > 2 when speaking to compatible agents |
+| >= 0.7 <= 1.10 | 2, 3. Will automatically use protocol > 2 when speaking to compatible agents |
+| 1.11+.         | 3                                                                            |
+
+-> **Note:** Regarding XDS protocol changes See https://developer.hashicorp.com/consul/docs/upgrading/upgrade-specific#envoy-xds-protocol-upgrades-1 for details.
```
Comment (blake) [inline website/content/docs/upgrading/compatibility.mdx:44]:
```suggestion
-> **Note:** Regarding XDS protocol changes See <https://developer.hashicorp.com/consul/docs/upgrading/upgrade-specific#envoy-xds-protocol-upgrades-1> for details.
```

Raw links in should be surrounded by angle brackets in Markdown.

**Pull request closed without merging**